gcloud-java-dns/src/main/java/com/google/gcloud/dns/Zone.java

```diff
@@ -68,166 +66,81 @@ public static Zone get(Dns dnsService, String zoneName, Dns.ZoneOption... option
-      deleted = dns.delete(zoneInfo.name());
-    }
-    return deleted;
+    return dns.delete(zoneInfo.name());
   }
 
   /**
-   * Lists all {@link DnsRecord}s associated with this zone. First searches for zone by ID and if
-   * not found then by name.
+   * Lists all {@link DnsRecord}s associated with this zone. The method searches for zone by name
+   * which must always be initialized.
    *
    * @param options optional restriction on listing and fields of {@link DnsRecord}s returned
    * @return a page of DNS records
    * @throws DnsException upon failure or if the zone is not found
-   * @throws NullPointerException if both zone ID and name are not initialized
    */
   public Page<DnsRecord> listDnsRecords(Dns.DnsRecordListOption... options) {
-    checkNameOrIdNotNull();
-    Page<DnsRecord> page = null;
-    if (zoneInfo.id() != null) {
-      page = dns.listDnsRecords(zoneInfo.id(), options);
-    }
-    if (page == null && zoneInfo.name() != null) {
-      // zone was not found by id or id is not set at all
-      page = dns.listDnsRecords(zoneInfo.name(), options);
-    }
-    return page;
+    return dns.listDnsRecords(zoneInfo.name(), options);
   }
 
   /**
-   * Submits {@link ChangeRequest} to the service for it to applied to this zone. First searches for
-   * the zone by ID and if not found then by name. Returns a {@link ChangeRequest} with
-   * server-assigned ID or {@code null} if the zone was not found.
+   * Submits {@link ChangeRequest} to the service for it to applied to this zone. The method
+   * searches for zone by name which must always be initialized.
    *
    * @param options optional restriction on what fields of {@link ChangeRequest} should be returned
    * @return ChangeRequest with server-assigned ID
    * @throws DnsException upon failure or if the zone is not found
-   * @throws NullPointerException if both zone ID and name are not initialized
    */
   public ChangeRequest applyChangeRequest(ChangeRequest changeRequest,
       Dns.ChangeRequestOption... options) {
-    checkNameOrIdNotNull();
     checkNotNull(changeRequest);
-    ChangeRequest updated = null;
-    if (zoneInfo.id() != null) {
-      updated = dns.applyChangeRequest(zoneInfo.id(), changeRequest, options);
-    }
-    if (updated == null && zoneInfo.name() != null) {
-      // zone was not found by id or id is not set at all
-      updated = dns.applyChangeRequest(zoneInfo.name(), changeRequest, options);
-    }
-    return updated;
+    return dns.applyChangeRequest(zoneInfo.name(), changeRequest, options);
   }
 
   /**
    * Retrieves an updated information about a change request previously submitted to be applied to
-   * this zone. First searches for the zone by ID and if not found then by name. Returns a {@link
-   * ChangeRequest} if found and {@code null} is the zone or the change request was not found.
+   * this zone. The method searches for zone by name which must always be initialized. Returns a
```

Comment:
I don't think we need to mention " The method searches for zone by name which must always be initialized."